src/pres/loader/PGNLoader.js

```diff
@@ -15,40 +15,33 @@ export default class PGNLoader extends React.Component {
 
     constructor(props) {
         super(props)
-        let selectedSite = new URLSearchParams(window.location.search).get("source")
-        let playerName = new URLSearchParams(window.location.search).get("playerName")
-        let color = new URLSearchParams(window.location.search).get("color") // 'white' | 'black'
-        // 'all' | 'rated' | 'casual' (corresponds to Filters.toggleRated)
-        let ratedMode = new URLSearchParams(window.location.search).get("ratedMode")
-        let fromDate = new URLSearchParams(window.location.search).get("fromDate") // YYYY-MM-DD
-        let toDate = new URLSearchParams(window.location.search).get("toDate") // YYYY-MM-DD
-        let timeControls = this.getTimeControls()
 
+        const urlSettings = this.props.urlSettings
         this.state = {
-            playerName: playerName?playerName:'',
-            site: selectedSite?selectedSite:'',
-            playerColor: color ? color : this.props.settings.playerColor,
-            isAdvancedFiltersOpen: false,
+            playerName: urlSettings.playerName,
+            site: urlSettings.site,
+            playerColor: urlSettings.color || this.props.settings.playerColor,
+            isAdvancedFiltersOpen: urlSettings.isAdvancedFiltersOpen,
```

Comment:
Previously, PGNLoader.state.isAdvancedFiltersOpen was (AFAICT, as a React novice) unused. Now I use it to pass urlSettings.isAdvancedFiltersOpen down to the Filters component.